src/server/async_store.ts

```diff
@@ -0,0 +1,15 @@
+import { BreadcrumbRecord, HoneybadgerStore } from "../core/types"
+
+let Store: HoneybadgerStore<{ context: Record<string, unknown>; breadcrumbs: BreadcrumbRecord[]; }>
+try {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const { AsyncLocalStorage } = require('async_hooks')
```

Comment:
We want to fallback to the `DefaultStore` if `AsyncLocalStorage` is not available.
I'm not 100% with my solution, I know this the CommonJS approach (using `require`), but I didn't find the equivalent with ESM (using `import`). Let me know if you have a better solution.

I tested this by enabling tests on Node v10 and the [tests started failing](https://github.com/honeybadger-io/honeybadger-js/runs/5260086192?check_suite_focus=true): `AsyncStore` was an instance `DefaultStore` and as expected, the assertions failed because the lambda handlers did not have isolated context.

Reply:
Yeah true, but we are supposed to have `import` statements at the top of the script, outside of any blocks (try/catch). That's why I used the `require`.